### ISSUE TITLE
Read historical prices (or general list data retrieval pattern)

### ISSUE THREAD

**Issue by schmidsi** [melonproject/protocol#482]
To visualize (share)price history we need to get the historical prices from the front-end which is exposed here:

https://github.com/melonproject/smart-contracts/blob/develop/src/pricefeeds/CanonicalPriceFeed.sol#L383-L390

On the Kovan test deployment at the time of writing we have ~20k price entries (`getHistoryLength()`). To get the complete history we need to loop through all of them and create a new JSON-RPC call for _every_ price entry to `getHistoryAt(uint id)`. This is almost impossible, even with a local node -> 20k requests.

Another possibility would be to model the response in a different way:

```
function getHistoryOfAsset(address asset) returns (uint[] timeStamps, uint[] prices)
```

Is there any advantage of the current implementation over this suggestion? What about risk engineering? In which format do we need this data to calculate for example volatility on chain?

**Comment by GauthamN** [comment]
Addressed in bb224735dd97ccb441ddd4c30bde30f5ed794e27

**Comment by schmidsi** [comment]
Awesome. Thanks a lot. Will it make into paros?
